scripts/deleteElasticRecords.py

```diff
@@ -28,26 +29,51 @@ def main():
 
     dbManager.createSession()
 
-    batchSize = 200
+    batchSize = 1000
+    esWorkUUIDS = deque()
+    psqlWorkUUIDTuples = deque()
     searchES = Search(index=os.environ['ELASTICSEARCH_INDEX'])
 
-    for work in searchES.query('match_all').scan():    
-        if dbManager.session.query(Work) \
-            .group_by(Work.uuid) \
-            .filter(Work.uuid == work.uuid) \
-            .yield_per(batchSize) == None:
-
-            try:
-                ESWork.delete(work.uuid)
-            except NotFoundError or ValueError or ConflictError:
-                if ValueError:
-                    print('Empty value')
-                elif ConflictError:
-                    print('Version number error')
-                else:
-                    print('Work not indexed, skipping')
+    for work in searchES.query('match_all').scan():
+        # Unpack the tuples to a flat array
+        # Convert the array of ElasticSearch UUIDs and the array of postgresql UUIDs to sets()
+        # Take the difference of of the ES set and the postgresql set (https://docs.python.org/3/library/stdtypes.html#frozenset.difference)
+        # what you have left should be a set of UUIDs that are in ES but not in psql
+            # that's what you should delete!
 
+        esWorkUUIDS.append(work.uuid) # Initialize the array outside of the loop here  
+        if len(esWorkUUIDS) >= batchSize:
+            findESOnlyWorks(esWorkUUIDS, psqlWorkUUIDTuples, dbManager, esManager)  
+            psqlWorkUUIDTuples = deque()
+            esWorkUUIDS = deque()
+
+    findESOnlyWorks(esWorkUUIDS, psqlWorkUUIDTuples, dbManager, esManager) #For any remainder ES works
+    
     dbManager.closeConnection()
 
+def findESOnlyWorks(esWorkUUIDS, psqlWorkUUIDTuples, dbManager, esManager):
+        for uuid in esWorkUUIDS:
+            uuidTuple = dbManager.session.query(Work.uuid) \
+                .filter(Work.uuid == uuid) \
+                .filter(Work.uuid.in_(esWorkUUIDS)).first()
+            if uuidTuple != None:
+                psqlWorkUUIDTuples.append(str(uuidTuple[0])) #PSQL work uuids are of type uuid unlike ES work uuids
+
+        setESWorkUUIDS = set(esWorkUUIDS) 
+        setPSQLWorkUUIDS = set(psqlWorkUUIDTuples)
+
+        onlyESWorkUUIDS = setESWorkUUIDS.difference(setPSQLWorkUUIDS) #Set should be empty
```

Comment:
What does the comment mean here? I would assume the set _shouldn't_ be empty!

Reply:
I was thinking of the cases where the ES records are also in PSQL which is usually most of the time when I wrote this comment. Most results I've seen in the terminal when printing onlyESWorkUUIDS is `set()`, but there are a few like
`{'084e9ab3-dfa4-4819-8536-0c31d5da5c46'}` that are the records we are looking for. I can see how you can be confused by my comment there.

Reply:
ah I see okay thanks! So we expect the set to usually be empty, that makes sense since we're looking for a relatively small number of records